gamemodes/terrortown/gamemode/shared/hud_elements/tttinfopanel/pure_skin_playerinfo.lua

```diff
@@ -107,8 +108,37 @@ if CLIENT then
 			else
 				text = L[self.roundstate_string[round_state]]
 			end
-
-			self:ShadowedText(text, "PureSkinRole", nx, ry, COLOR_WHITE, TEXT_ALIGN_LEFT, TEXT_ALIGN_CENTER)
+			
+			--calculate the scale multplier for role text
+			surface.SetFont("PureSkinRole")
+			local role_text_width = surface.GetTextSize(string.upper(text))
+			local role_scale_multiplier = (w - lpw - 2 * pad) / role_text_width 
+			if calive and cactive and secondaryRoleInformationFunc then
+				local secInfoTbl = secondaryRoleInformationFunc()
+				
+				if secInfoTbl and secInfoTbl.text then
+					surface.SetFont("PureSkinBar")
+					local sri_text_width = surface.GetTextSize(string.upper(secInfoTbl.text))
+					role_scale_multiplier = (w - sri_text_width - lpw - 2 * pad - 3 * sri_text_width_padding) / role_text_width 
+				end
+			end		
+			 
+			role_scale_multiplier = math.Clamp(role_scale_multiplier, 0.55, 0.92)
+			
+			
+			--create scaling matrix for the text
+			local mat = Matrix()
+			mat:Translate( Vector( nx, ry ) )
+			mat:Scale( Vector( role_scale_multiplier, role_scale_multiplier, role_scale_multiplier ) )
+			mat:Translate( -Vector( nx, ry ) )
+			
```

Comment:
We definitely need a performance test. A way to speed up theese pushes would be to create a table with a list of texts and just pushing the render filter once

Reply:
I don't really think that this would boost the performance, since there are no complex computations behind push and pop. Instead it would decrease the readability and therefore is in my opinion negligible.

Reply:
Since the performance of "ShadowedText" is the bottleneck in the function, I'm completely with @saibotk.